RELEASING.md

```diff
@@ -0,0 +1,79 @@
+# Releasing knative/pkg
+
+We release the components of Knative ever 6 weeks.  All of these components must
+be moved to the latest "release" of the knative/pkg shared library prior to each
+release, but likely this should happen incrementally over each milestone.
+
+## Release Process.
+
+### Step #1: Monday the week prior to each release.
+
+Early the week prior to the Knative release, each of the downstream repositories
+should stage a `[WIP]` Pull Request that advances the knative/pkg dependency to
+the latest commit:
+
+`Gopkg.toml` should look like:
+
+```toml
+[[override]]
+  name = "github.com/knative/pkg"
+  branch = "master"
+```
+
+Then the following is run:
+
+```shell
+dep ensure -update github.com/knative/pkg
+```
+
+If problems are found, they are addressed and the update is merged, and this
+process repeats until knative/pkg can be cleanly updated without any changes.
```

Comment:
There should probably be something about the repos need to keep checking (or can some sort of notification be set up?). I'm thinking of the scenario where, say eventing repo does this on Monday morning, everything looks great and then some other repo has an issue, requiring an update to pkg, so now every repo needs to grab the newer version. Saying you have to manually check the pkg repo is fine, but point being is that there's some on-going work to keep checking and re-testing.

Reply:
We now have automation that updates the list of repos above to the latest commit in master daily like the rest, so the above basically devolves to making sure that is working, but it's written as if it's the only game in town (because I wrote the automation after putting this together to try and make this easier!).